Library/Homebrew/os/mac/xcode.rb

```diff
@@ -234,6 +234,30 @@ def detect_version_from_clang_version
       def default_prefix?
         prefix.to_s == "/Applications/Xcode.app/Contents/Developer"
       end
+
+      sig { params(swift_version: String).returns(String) }
+      def minimum_version_for_swift(swift_version)
+        # https://swiftversion.net
+        case swift_version
```

Comment:
Which of these versions do we have software depending on?

Reply:
Oldest I've seen is Xcode 8 (Swift 3). Though that's probably because we've perhaps purged everything older than that.